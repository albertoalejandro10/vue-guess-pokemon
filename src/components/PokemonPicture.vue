<template>
  <div class="pokemon-container">
    <img v-if="showPokemon" :src="getPokemon" class="fade-in" alt="pokemon" />
    <img v-else :src="getPokemon" class="hidden-pokemon" alt="pokemon" />
  </div>
</template>
<script>
export default {
  props: {
    pokemonId: {
      type: Number,
      required: true,
    },
    showPokemon: {
      type: Boolean,
      required: true,
      default: false,
    },
  },
  computed: {
    getPokemon() {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`
    },
  },
}
</script>
<style scoped>
/* Pokemon Picture */
.pokemon-container {
  height: 200px;
  width: 100%;
  display: grid;
  grid-template-columns: 1fr;
}

img {
  height: 200px;
  grid-area: 1/1;
  width: 100%;
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}
.hidden-pokemon {
  filter: brightness(0);
}
</style>
