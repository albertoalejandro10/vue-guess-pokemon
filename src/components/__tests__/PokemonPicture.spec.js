import { shallowMount } from "@vue/test-utils"
import { describe, expect, it } from "vitest"
import PokemonPicture from "@/components/PokemonPicture.vue"

describe("PokemonPicture component", () => {
  it("Debe hacer match con el snapshot", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 1,
        showPokemon: false,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it("Debe de mostrar la imagen oculta y el pokemon 100", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: false,
      },
    })
    const [img1, img2] = wrapper.findAll("img")
    expect(img1.exists()).toBeTruthy()
    expect(img2).toBeUndefined()
    expect(img1.classes("hidden-pokemon")).toBeTruthy()
    const expectedURL =
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"
    expect(img1.attributes("src")).toBe(expectedURL)
  })
  it("Debe de mostrar el pokemon si showPokemon: true", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: true,
      },
    })
    const img1 = wrapper.find("img")
    expect(img1.exists()).toBeTruthy()
    expect(img1.classes("hidden-pokemon")).toBeFalsy()
    expect(img1.classes("fade-in")).toBeTruthy()
  })
})
