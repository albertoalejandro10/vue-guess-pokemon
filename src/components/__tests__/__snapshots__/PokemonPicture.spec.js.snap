// Vitest Snapshot v1

exports[`PokemonPicture component > Debe hacer match con el snapshot 1`] = `"<div class=\\"pokemon-container\\" data-v-34cacee9=\\"\\"><img src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg\\" class=\\"hidden-pokemon\\" alt=\\"pokemon\\" data-v-34cacee9=\\"\\"></div>"`;
